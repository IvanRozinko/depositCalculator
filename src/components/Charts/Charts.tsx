import React from 'react'
import { Box } from '@material-ui/core';
import { VerticalRectSeries, FlexibleWidthXYPlot } from 'react-vis';
import InfoRow from '../InfoRow/InfoRow';

import { decimalFormatter } from '../../helpers/helpers'

interface Props {
  sum: number,
  monthlyTotal: number,
  profit: number
}

function Charts({ sum, monthlyTotal, profit }: Props) {

  const leftChartX = { x0: 0, x: 4 };
  const rightChartX = { x0: 5, x: 9 };
  const totalIncome = sum + monthlyTotal;
  const totalRangeY = totalIncome + profit;

  const rangeY = (value: number) => totalRangeY / 100 * value;
  const firstBar = [{...leftChartX, y: rangeY(sum), y0: 0}, {...rightChartX, y: rangeY(sum), y0: 0}];
  const secondBar = [{...leftChartX, y: rangeY(monthlyTotal), y0: 0}, {...rightChartX, y: rangeY(monthlyTotal), y0: 0}];
  const thirdBar = [{...leftChartX, y: 0, y0: 0}, {...rightChartX, y: rangeY(profit), y0: 0}];

  return (
    <Box display='flex' flexDirection='column'>
      <InfoRow label='Вклав' value='Отримано' />
      <InfoRow 
        label={decimalFormatter(totalIncome)} 
        value={decimalFormatter(totalIncome + profit)} 
        valueBold 
      />
      <FlexibleWidthXYPlot height={300} stackBy="y">
        <VerticalRectSeries data={firstBar} color='darkgray' />
        <VerticalRectSeries data={secondBar} color='#35322B' />
        <VerticalRectSeries data={thirdBar}  color='#3f51b5' />
      </FlexibleWidthXYPlot>
    </Box>
  )
}

export default Charts
