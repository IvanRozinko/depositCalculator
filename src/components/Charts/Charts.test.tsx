import Charts from './Charts';
import { shallow } from 'enzyme';

describe('<Chats /> :', () => {
  const defaultProps = {
    sum: 100,
    monthlyTotal: 300,
    profit: 20
  }

  const wrapper = shallow(
    <Charts {...defaultProps} />
  )

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})