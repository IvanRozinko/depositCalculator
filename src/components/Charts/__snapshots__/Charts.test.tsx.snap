// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chats /> : should render correctly 1`] = `
<Styled(MuiBox)
  display="flex"
  flexDirection="column"
>
  <InfoRow
    centered={true}
    label="Вклав"
    value="Отримано"
  />
  <InfoRow
    centered={true}
    label="400"
    value="420"
    valueBold={true}
  />
  <FlexibleXYPlot
    height={300}
    stackBy="y"
  >
    <VerticalRectSeries
      className=""
      color="darkgray"
      data={
        Array [
          Object {
            "x": 4,
            "x0": 0,
            "y": 420,
            "y0": 0,
          },
          Object {
            "x": 9,
            "x0": 5,
            "y": 420,
            "y0": 0,
          },
        ]
      }
      stack={false}
      style={Object {}}
    />
    <VerticalRectSeries
      className=""
      color="#35322B"
      data={
        Array [
          Object {
            "x": 4,
            "x0": 0,
            "y": 1260,
            "y0": 0,
          },
          Object {
            "x": 9,
            "x0": 5,
            "y": 1260,
            "y0": 0,
          },
        ]
      }
      stack={false}
      style={Object {}}
    />
    <VerticalRectSeries
      className=""
      color="#3f51b5"
      data={
        Array [
          Object {
            "x": 4,
            "x0": 0,
            "y": 0,
            "y0": 0,
          },
          Object {
            "x": 9,
            "x0": 5,
            "y": 84,
            "y0": 0,
          },
        ]
      }
      stack={false}
      style={Object {}}
    />
  </FlexibleXYPlot>
</Styled(MuiBox)>
`;
